launchdarkly/resource_environment.go

```diff
@@ -0,0 +1,146 @@
+package launchdarkly
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	"sync"
+)
+
+// Since we cannot delete the last environment in a project, we use a hack with a temporary dummy
+// environment that is created when we want to delete the last one. To avoid race conditions when
+// deleting multiple resources (terraform by default runs such operations concurrently), we need
+// to use a mutex.
+var environmentMutex = &sync.Mutex{}
+
+func resourceEnvironment() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceEnvironmentCreate,
+		Read:   resourceEnvironmentRead,
+		Update: resourceEnvironmentUpdate,
+		Delete: resourceEnvironmentDelete,
+
+		Schema: map[string]*schema.Schema{
+			"project_key": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"key": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"color": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceEnvironmentCreate(d *schema.ResourceData, m interface{}) error {
+	client := m.(Client)
+
+	environmentMutex.Lock()
+	defer environmentMutex.Unlock()
+
+	project := d.Get("project_key").(string)
```

Comment:
At this point you could make a struct to store that payload right?
Would make it marshallable/unmarshallable with type validation.

Reply:
Well I tried, but failed. Is that support to be possible at all?

Reply:
More precisely, the call would run but all the fields would be left empty.

If you tell me it's supposed to work I'll have a stab at it again for sure.

Reply:
Yes that's what I meant :) sorry should've given an example.

Reply:
I did try those, it didn't seem to work better though.

Will try again.

Reply:
There. No more using maps for JSON. Satisfied?